src/Dapr.Actors/Runtime/Actor.cs

```diff
@@ -328,9 +291,39 @@ protected async Task UnregisterTimerAsync(IActorTimer timer)
         protected async Task UnregisterTimerAsync(string timerName)
         {
             await ActorRuntime.DaprInteractor.UnregisterTimerAsync(this.actorTypeName, this.Id.ToString(), timerName);
-            if (this.timers.ContainsKey(timerName))
+        }
+
+        private void ValidateTimerCallback(string callback)
+        {
+            var actorTypeName = this.Host.ActorTypeInfo.ActorTypeName;
+            var actorType = this.Host.ActorTypeInfo.ImplementationType;
+
+            // GetMethod will throw an AmbiguousMatchException if more than one methods are found with the same name
```

Comment:
Is the error message clear when that happens?

Reply:
I caught the exception and threw with a more understandable error message